### DISCUSSION TITLE
Nav Items for CP

### DISCUSSION THREAD

**Discussion by HazemMohamed252** [statamic/cms#7643] [Q&A]
Nav::extend(function ($nav) {
            $nav->create('General Settings')
                    ->section('System Settings')
                    ->url('generalsettings')
                    ->can('View general settings')
                    ->children([
                        $nav->item('Email Settings')->url('emailsettings')->can('View email settings'),
                        $nav->item('SMS Settings')->url('smssettings')->can('View sms settings'),
                        $nav->item('Onboardings')->url('onboardings')->can('View onboardings'),
                        $nav->item('Value Chains')->url('valuechains')->can('View value chains'),
                        $nav->item('Produces')->url('produces')->can('View produces'),
                        $nav->item('Interests And Activities')->url('activitytypes')->can('View activitytypes'),
                        $nav->item('Irrigations Types')->url('irrigations')->can('View irrigations'),
                        $nav->item('Fertilizations Types')->url('fertilizations')->can('View fertilizations'),
                        $nav->item('User Activities')->url('clienttypes')->can('View client types'),
                        $nav->item('Sections Ads')->url('sectionsads')->can('View sections ads'),
                        $nav->item('About Bashaier')->url('aboutbashaier')->can('View about bashaier')
            ]);
        });

Please i just created nav with childrens but when i click on any children the menu is collapsed again. I need when i click on children the menu still opened and the children is active. Please ASAP

**Comment by duncanmcclean** [comment]
I'm not sure about how to fix your issue but would you be able to surround your code in backticks (`) so it's rendered in a code block?

**Comment by jasonvarga** [comment]
Make the child urls actual children in the url too.

If the parent is `/generalsettings`, make the children `/generalsettings/emailsettings`, etc.

**Reply by HazemMohamed252** [reply]
I have another issue. After i did this solution for example : i create generalsettings as parent then created /generalsettings/emailsettings.
Then when i gave permission for the child emailsettings without giving permission for the parent the sidemenu not appears.
This is my code:
Nav::extend(function ($nav) {
            $nav->create('Questions Categories')
                    ->section('Ask the expert')
                    ->url('questionscategories')
                    ->can('View questions categories')
                    ->children([
                        $nav->item('Questions')->url('questionscategories/questions')->can('View questions')
            ]);
        });
        
        In above example Questions Categories has childern questions. In the CP i gave permission for questions ONLY without questions categories permission so the sidemenu not appear. Please how to fix this.

**Reply by jasonvarga** [reply]
If you don't have permission to the parent then it makes sense that it wouldn't appear.
